extensions/compose/src/extension.ts

```diff
@@ -284,9 +284,73 @@ async function registerCLITool(composeDownload: ComposeDownload, detect: Detect)
       installationSource,
     });
   }
+
+  // register the updater to allow users to upgrade/downgrade their cli
+  let releaseToUpdateTo: ComposeGithubReleaseArtifactMetadata | undefined;
+  let releaseVersionToUpdateTo: string | undefined;
+
+  let latestVersionAsset: ComposeGithubReleaseArtifactMetadata | undefined;
+  try {
+    latestVersionAsset = await composeDownload.getLatestVersionAsset();
+  } catch (error) {
+    console.log('Error when loading compose latest release meta.', error);
+  }
+
+  const latestVersion = latestVersionAsset ? removeVersionPrefix(latestVersionAsset.tag) : undefined;
+
+  const update = {
+    version: latestVersion,
+    selectVersion: async (): Promise<string> => {
+      const selected = await composeDownload.promptUserForVersion(binaryVersion);
+      releaseToUpdateTo = selected;
+      releaseVersionToUpdateTo = removeVersionPrefix(selected.tag);
+      return releaseVersionToUpdateTo;
+    },
+    doUpdate: async (): Promise<void> => {
+      if (!binaryVersion || !binaryPath) {
+        throw new Error(`Cannot update ${composeCliName}. No cli tool installed.`);
+      }
+      if (!releaseVersionToUpdateTo) {
+        // update request to latest version
+        if (latestVersionAsset) {
+          releaseToUpdateTo = latestVersionAsset;
+          releaseVersionToUpdateTo = latestVersion;
+        }
+      }
+      if (!releaseToUpdateTo || !releaseVersionToUpdateTo) {
+        throw new Error(`Cannot update ${binaryInfo?.path} version ${binaryVersion}. No release selected.`);
+      }
+      if (!binaryInfo?.updatable) {
+        throw new Error(
+          `Cannot update ${binaryInfo?.path} version ${binaryVersion} to ${releaseVersionToUpdateTo} as it was not installed by podman-desktop`,
+        );
+      }
+
+      // download, install system wide and update cli version
+      await composeDownload.download(releaseToUpdateTo);
+      // get the binary in the extension folder
+      const storagePath = await detect.getStoragePath();
+      binaryPath = await installBinaryToSystem(storagePath, composeCliName);
```

Comment:
try {
 // try to install system wide
} catch {
// log the error
 } finally {
// update the composeCliTool with either the path from the extension folder or the system wide if install success
}

Reply:
Same as previous comment.